benchmark/assert/deepequal-buffer.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+const common = require('../common.js');
+const assert = require('assert');
+const bench = common.createBenchmark(main, {
+  n: [1e3],
+  len: [1e2],
+  strict: ['strict', 'nonstrict']
+});
+
+function main(conf) {
+  const n = +conf.n;
+  const len = +conf.len;
+  var i;
+
+  const data = Buffer.allocUnsafe(len);
+  const actual = Buffer.alloc(len);
+  const expected = Buffer.alloc(len);
+  data.copy(actual);
+  data.copy(expected);
+
+  if (conf.strict === 'strict') {
```

Comment:
It's also possible to simplify this further by doing something like:

```js
var method;
switch (conf.strict) {
  case 'strict':
    method = assert.deepStrictEqual;
    break;
  case 'nonstrict':
    method = assert.deepEqual;
    break;
  default:
    throw new Error('...');
}
bench.start();
for (i = 0; i < n; ++i)
  method(actual, expected);
bench.end(n);
```

Or even:

```js
const methods = {
  'strict': assert.deepStrictEqual,
  'nonstrict': assert.deepEqual
};
const method = methods[conf.strict];
if (!method) throw new Error('...');
bench.start();
for (i = 0; i < n; ++i)
  method(actual, expected);
bench.end(n);
```